src/openrct2/editor.c

```diff
@@ -528,3 +529,174 @@ bool editor_check_park()
 
 	return true;
 }
+
+void game_command_edit_scenario_options(sint32* eax, sint32* ebx, sint32* ecx, sint32* edx, sint32* esi, sint32* edi, sint32* ebp)
+{
+	if (*ebx & GAME_COMMAND_FLAG_APPLY) {
+		switch (*ecx) {
+		case EDIT_SCENARIOOPTIONS_SETNOMONEY:
+			if (gScreenFlags & SCREEN_FLAGS_SCENARIO_EDITOR) {
+				if (*edx != 0) {
+					gParkFlags |= PARK_FLAGS_NO_MONEY_SCENARIO;
+				}
+				else {
+					gParkFlags &= PARK_FLAGS_NO_MONEY_SCENARIO;
+				}
+			}
+			else {
+				if (*edx != 0) {
+					gParkFlags |= PARK_FLAGS_NO_MONEY;
+				}
+				else {
+					gParkFlags &= ~PARK_FLAGS_NO_MONEY;
+				}
+				// Invalidate all windows that have anything to do with finance
+				window_invalidate_by_class(WC_RIDE);
+				window_invalidate_by_class(WC_PEEP);
+				window_invalidate_by_class(WC_PARK_INFORMATION);
+				window_invalidate_by_class(WC_FINANCES);
+				window_invalidate_by_class(WC_BOTTOM_TOOLBAR);
+				window_invalidate_by_class(WC_TOP_TOOLBAR);
+			}
+			break;
+		case EDIT_SCENARIOOPTIONS_SETINITIALCASH:
+			gInitialCash = max(MONEY(0,00), min(MONEY(1000000,00), *edx));
+			gCashEncrypted = ENCRYPT_MONEY(gInitialCash);
+			window_invalidate_by_class(WC_FINANCES);
+			window_invalidate_by_class(WC_BOTTOM_TOOLBAR);
+			break;
+		case EDIT_SCENARIOOPTIONS_SETINITIALLOAN:
+			gBankLoan = max(MONEY(0,00), min(MONEY(5000000,00), *edx));
+			gMaxBankLoan = max(gBankLoan, gMaxBankLoan);
+			window_invalidate_by_class(WC_FINANCES);
+			break;
+		case EDIT_SCENARIOOPTIONS_SETMAXIMUMLOANSIZE:
+			gMaxBankLoan = max(MONEY(0,00), min(MONEY(5000000,00), *edx));
+			gBankLoan = min(gBankLoan, gMaxBankLoan);
+			window_invalidate_by_class(WC_FINANCES);
+			break;
+		case EDIT_SCENARIOOPTIONS_SETANNUALINTERESTRATE:
+			gBankLoanInterestRate = max(0, min(80, *edx));
+			window_invalidate_by_class(WC_FINANCES);
+			break;
+		case EDIT_SCENARIOOPTIONS_SETFORBIDMARKETINGCAMPAIGNS:
+			if (*edx != 0) {
```

Comment:
Ironically theres no text notification in the marketing window. When you go to initiate a new campaign, it brings up an error message. that's the only time it checks to see if marketing campaigns are forbidden.